lib/handlers/publish.js

```diff
@@ -25,96 +25,143 @@ const validReasonCodes = [0, 16, 128, 131, 135, 144, 145, 151, 153]
-            client.emit('error', new Error('Received Topic Alias is out of range'))
-            return
-          }
-        } else {
-          if (client.topicAliasRecv.put(topic, alias)) {
-            client.log('handlePublish :: registered topic: %s - alias: %d', topic, alias)
-          } else {
-            client.log('handlePublish :: topic alias out of range. alias: %d', alias)
-            client.emit('error', new Error('Received Topic Alias is out of range'))
-            return
-          }
-        }
-      }
-    }
-    client.log('handlePublish: qos %d', qos)
-    switch (qos) {
-      case 2: {
-        options.customHandleAcks(topic, message, packet, function (error, code) {
-          if (!(error instanceof Error)) {
-            code = error
-            error = null
-          }
-          if (error) { return client.emit('error', error) }
-          if (validReasonCodes.indexOf(code) === -1) { return client.emit('error', new Error('Wrong reason code for pubrec')) }
-          if (code) {
-            client._sendPacket({ cmd: 'pubrec', messageId, reasonCode: code }, done)
-          } else {
-            client.incomingStore.put(packet, function () {
-              client._sendPacket({ cmd: 'pubrec', messageId }, done)
-            })
-          }
-        })
-        break
-      }
-      case 1: {
-        // emit the message event
-        options.customHandleAcks(topic, message, packet, function (error, code) {
-          if (!(error instanceof Error)) {
-            code = error
-            error = null
-          }
-          if (error) { return client.emit('error', error) }
-          if (validReasonCodes.indexOf(code) === -1) { return client.emit('error', new Error('Wrong reason code for puback')) }
-          if (!code) { client.emit('message', topic, message, packet) }
-          client.handleMessage(packet, function (err) {
-            if (err) {
-              return done && done(err)
-            }
-            client._sendPacket({ cmd: 'puback', messageId, reasonCode: code }, done)
-          })
-        })
-        break
-      }
-      case 0:
-        // emit the message event
-        client.emit('message', topic, message, packet)
-        client.handleMessage(packet, done)
-        break
-      default:
-        // do nothing
-        client.log('handlePublish: unknown QoS. Doing nothing.')
-        // log or throw an error about unknown qos
-        break
-    }
+	client.log('handlePublish: packet %o', packet)
+	done = typeof done !== 'undefined' ? done : client.nop
+	let topic = packet.topic.toString()
+	const message = packet.payload
+	const { qos } = packet
+	const { messageId } = packet
+	const { options } = client
+	if (client.options.protocolVersion === 5) {
+		let alias
+		if (packet.properties) {
+			alias = packet.properties.topicAlias
+		}
+		if (typeof alias !== 'undefined') {
+			if (topic.length === 0) {
+				if (alias > 0 && alias <= 0xffff) {
+					const gotTopic =
```

Comment:
nit: I feel like the newline here is not really useful. Maybe increase the max line length setting?

Reply:
Line length increesed to 100 (default is 80)

Reply:
I think I would revert this as I found out: https://prettier.io/docs/en/options.html#print-width